core/actions/assertion.ts

```diff
@@ -243,12 +242,12 @@ export class Assertion extends ActionBuilder<dataform.Assertion> {
    * assertion.
    */
   public database(database: string) {
-    setNameAndTarget(
-      this.session,
-      this.proto,
-      this.proto.target.name,
-      this.proto.target.schema,
-      database
+    this.proto.target = this.applySessionToTarget(
+      dataform.Target.create({ ...this.proto.target, database }),
+      this.session.projectConfig,
+      this.proto.fileName,
+      true,
```

Comment:
Having two booleans here is very confusing to the reader. Ideally I would change the method API to accept an object which contains all the fields, it will allow to pass named parameters.

But to make it simpler could we just add comments like:

```typescript
public database(database: string) {
    this.proto.target = this.applySessionToTarget(
      dataform.Target.create({ ...this.proto.target, database }),
      this.session.projectConfig,
      this.proto.fileName,
      /* validateTarget= */ true,
      /* useDefaultAssertionDataset= */ true
    );
```

I know it is annoying to write it but will be much easier to understand the code afterwards.

Reply:
Fairs, comments are the worst way to do this though - have made it an object so that naming is required